gpt4all-backend/llmodel.cpp

```diff
@@ -98,3 +98,17 @@ LLModel *LLModel::construct(const std::string &modelPath, std::string buildVaria
     // Return final instance
     return fres;
 }
+
+void LLModel::prompt(const std::string &prompt_, std::function<bool (int32_t)> promptCallback, std::function<bool (int32_t, const std::string &)> responseCallback, std::function<bool (bool)> recalculateCallback, PromptContext &ctx) {
```

Comment:
In the future, we're going to need the prompt and response callbacks to both include token id, token string, and logit. In addition, the prompt callback should indicate whether a particular token is part of the prompt template or not as well.

Reply:
Should I implement this inside this PR?